### ISSUE TITLE
Compiler segfault when storing the type of a first class function directly into a type variable

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#13209]
### Summary of Problem
The `prune` pass encounters a segfault when a first class function's type is stored directly into a type variable instead of storing the fcf into a variable and taking the type of that.  This seems to be due to accessing an outdated portion of the virtualMethodTable (I believe we do not search through the virtualMethodTable to remove cleaned up AST nodes).

Encountered by @krishnadey30 when writing UnitTest stuff

### Steps to Reproduce

**Source Code:**

```chapel
proc foo() {
  writeln("in foo");
}

// If this instead reads:
// var tempfcf = foo;
// type argType = tempfcf.type;
// then the compiler does not segfault.
type argType = foo.type;
writeln(argType: string);
```

**Compile command:**
`chpl foo.chpl`

**Associated Future Test(s):**
[`test/functions/firstClassFns/storeTheType.chpl`](https://github.com/chapel-lang/chapel/blob/master/test/functions/firstClassFns/storeTheType.chpl) #13222 

### Configuration Information

- Output of `chpl --version`: 1.20 (pre-release)

**Comment by lydia-duncan** [comment]
This can sporadically succeed and reach execution.

**Comment by daviditen** [comment]
Fixed by #14218.
